snippets/chapel.json

```diff
@@ -8,9 +8,128 @@
     ],
     "description": "a simple parallel loop"
   },
+  "forall-single": {
+    "prefix": "forall",
+    "body": [
+      "forall $1 in $2 do $3;"
+    ],
+    "description": "a simple parallel loop -- single line"
+  },
+  "for": {
+    "prefix": "for",
+    "body": [
+      "for $1 in $2 {",
+      "  $3",
+      "}"
+    ],
+    "description": "a simple serial loop"
+  },
+  "for-single": {
+    "prefix": "for",
+    "body": [
+      "for $1 in $2 do $3;"
+    ],
+    "description": "a simple serial loop -- single line"
+  },
+  "while": {
+    "prefix": "while",
+    "body": [
+      "while $1 {",
+      "  $2",
+      "}"
+    ],
+    "description": "while construct"
+  },
+  "while-single": {
+    "prefix": "while",
+    "body": [
+      "while $1 do $2;"
+    ],
+    "description": "while construct -- single line"
+  },
+  "dowhile": {
+    "prefix": "dowhile",
+    "body": [
+      "do {",
+      "  $1",
+      "} while $2;"
+    ],
+    "description": "while construct"
+  },
+  "if": {
+    "prefix": "if",
+    "body": [
+      "if $1 {",
+      "  $2",
+      "}"
+    ],
+    "description": "if statement"
+  },
+  "if-single": {
+    "prefix": "if",
+    "body": [
+      "if $1 then $2;"
+    ],
+    "description": "if statement -- single line"
+  },
+  "ifelse": {
+    "prefix": "ifelse",
+    "body": [
+      "if $1 {",
+      "  $2",
+      "} else {",
+      "  $3",
+      "}"
+    ],
+    "description": "if else statement"
+  },
+  "ifelse-single": {
+    "prefix": "ifelse",
+    "body": [
+      "if $1 then $2; else $3;"
+    ],
+    "description": "if else statement -- single line"
+  },
+  "select": {
+    "prefix": "select",
+    "body": [
+      "select $1 {",
+      "  when $2 do $3;",
+      "  otherwise $4",
+      "}"
+    ],
+    "description": "select block"
+  },
+  "proc": {
+    "prefix": "proc",
+    "body": [
+      "proc $1($2) {",
+      "  $3",
+      "}"
+    ],
+    "description": "procedure definition"
+  },
+  "proc-single": {
+    "prefix": "proc",
+    "body": [
+      "proc $1($2) return $3;"
```

Comment:
I believe the doc you referenced is out of date, single return procedures are no longer supported by Chapel. You could change this to `proc $1($2) do $3;`, as Chapel does support single statement procedures using the `do` keyword

Reply:
Yes it should be updated, after seeing this yesterday I opened https://github.com/chapel-lang/chapel/pull/24747 to fix this case